sorting/tim_sort.c

```diff
@@ -0,0 +1,110 @@
+#include<stdio.h>
+#include<stdlib.h>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c
#include <stdio.h>     /// for IO operations
#include <assert.h>    /// for assert
```